homeassistant/components/dsmr_reader/config_flow.py

```diff
@@ -0,0 +1,43 @@
+"""Config flow to configure DSMR Reader."""
+from __future__ import annotations
+
+from collections.abc import Awaitable
+import logging
+from typing import Any
+
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.config_entry_flow import DiscoveryFlowHandler
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def _async_has_devices(_: HomeAssistant) -> bool:
+    """Return true as this integration doesn't support any real devices."""
+    return True
```

Comment:
This should return False if the data isn't there. I don't have MQTT installed, but manually configuring the user flow would let me set it up.

Reply:
That would indeed be nicer. The challenge with that is that I'd have to wait for a couple of seconds to await whether any dsmr topics are actually present. I don't think there's a quick and easy way to figure out whether the data is present. Unless I'm overlooking something, of course.

Reply:
`return "mqtt" in hass.config.components` should be a reasonable check here.

Reply:
Ah, now I see what you mean. I've done some testing and that statement always seems to return true, even in a completely fresh installation. I've gone for `hass.services.has_service(domain="mqtt", service="publish")` instead now, as that can properly distinguish between MQTT being known and being configured.

I've also had to add a string for `config.abort.no_devices_found`. I went for a custom string as "No devices found on the network" doesn't really give a clue on what is going wrong.